sound/soc/sof/core.c

```diff
@@ -238,6 +239,12 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 		goto fw_trace_err;
 	}
 
+	ret = sof_register_clients(sdev);
```

Comment:
I hesitated for a moment, but I think yes, you probably first want to register the machine driver and only then add any optional clients

Reply:
Eventually we will be moving the machine driver also to client, but let me take a look.
Fwiw, even when I had the client registration after the machine driver registering the probes card would be instantiated first. I think there is a delayed work or something involved on the machine driver side or the the components registration is out of bounds via topology parsing and stuff?

Reply:
Ah, you were saying the same thing ;)
What I have noticed while looking at it is that the client un-registration was in a wrong place, now fixed locally.

Thanks!